otx/core/config/manager.py

```diff
@@ -0,0 +1,110 @@
+"""Auto Configuration Manager ."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+from typing import Any, Dict
+
+import mmcv
+from datumaro.components.dataset import Dataset
+from datumaro.components.dataset_base import IDataset
+
+from otx.core.data.manager.dataset_manager import DatasetManager
+
+
+class ConfigManager:
+    """Auto configuration manager that could set the proper configuration.
+
+    Currently, it only supports the small amount of functions.
+    * Data format detection
+    * Task type detection
+    * Write the data to the workspace
+    * Write the data configuration to the workspace
+
+    However, it will supports lots of things in the near future.
+    * Automatic train type detection (Supervised, Self, Semi)
+    * Automatic resource allocation (num_workers, HPO)
+
+    """
+
+    def __init__(self):
+        self.task_data_dict = {
+            "ACTION_CLASSIFICATION": ["multi-cvat"],
+            "ACTION_DETECTION": ["multi-cvat"],
+            "ANOMALY_CLASSIFICATION": ["mvtec"],
+            "ANOMALY_DETECTION": ["mvtec"],
+            "ANOMALY_SEGMENTATION": ["mvtec"],
+            "CLASSIFICATION": ["imagenet"],
+            "DETECTION": ["coco", "voc", "yolo"],
+            "INSTANCE_SEGMENTATION": ["coco", "voc"],
+            "SEGMENTATION": ["common_semantic_segmentation", "voc", "cityscapes", "ade20k2017", "ade20k2020"],
+        }
+
+    def get_task_type(self, data_format: str) -> str:
+        """Detect task type.
+
+        For some datasets (i.e. COCO, VOC, MVTec), can't be fully automated.
+        Because those datasets have several format at the same time.
+        (i.e. for the COCO case, object detection and instance segmentation annotations coexist)
+        In this case, the task_type will be selected to default value.
+
+        For action tasks, currently action_classification is default.
+
+        If Datumaro supports the Kinetics, AVA datasets, MVTec, _is_cvat_format(), _is_mvtec_format()
+        functions will be deleted.
+        """
+
+        task = ""
+        if data_format == "multi-cvat":
```

Comment:
why do we check it here if we store this in `self.task_data_dict` and  it can be checked in the loop below?